### PR TITLE
Set extAuth authority to hostname 

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

to make authority aligns with otel and trace reporting.

Otherwise, we could meet https://istio.slack.com/archives/C382V8Q92/p1691034317554989?thread_ts=1691033025.465709&cid=C382V8Q92

### CODE REVIEW

File: pilot/pkg/security/authz/builder/testdata/http/custom-grpc-provider-no-namespace-out2.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ typedConfig:
         prefix: istio-ext-authz
   grpcService:
     envoyGrpc:
-      authority: outbound_.9000_._.my-custom-ext-authz.foo.svc.cluster.local
+      authority: my-custom-ext-authz.foo.svc.cluster.local
```
Comment (howardjohn) [inline pilot/pkg/security/authz/builder/testdata/http/custom-grpc-provider-no-namespace-out2.yaml:14]: Doesn't it matter if its mTLS or not?

For mTLS we need the old name. For non-mtls this name is fine

Comment (howardjohn) [reply pilot/pkg/security/authz/builder/testdata/http/custom-grpc-provider-no-namespace-out2.yaml:14]: Oh nevermind that is for SNI, this is authority

File: pilot/pkg/security/authz/builder/extauthz.go [hunk 2]
```diff
@@ -249,16 +249,8 @@ func generateHTTPConfig(hostname, cluster string, status *envoytypev3.HttpStatus
 		},
 	}
 	allowedHeaders := generateHeaders(config.IncludeRequestHeadersInCheck)
-	if allowedHeaders == nil {
-		// IncludeHeadersInCheck is deprecated, only use it if IncludeRequestHeadersInCheck is not set.
```
[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pilot/pkg/security/authz/builder/extauthz.go:253]: Kind of understand, istio api v1beta1 is almost the same as v1alpha3. I am concerned we would take all the legacy burdens when we evolve to V1.

Will restore this though.

**Review by howardjohn — APPROVED** [review]
Release note would be good

**Pull request merged**